### ISSUE TITLE
False positive `reportOptionalMemberAccess`

### ISSUE THREAD

**Issue by Molkree** [microsoft/pylance-release#2549]
## Environment data

-   Language Server version: v2022.4.0
-   OS and version: Win10
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.10

## Expected behaviour

No error.

## Actual behaviour

`"next" is not a known member of "None"`

## Code Snippet / Additional information

```python
from __future__ import annotations


class ListNode:
    def __init__(self, val: int = 0, next: ListNode | None = None):
        self.val = val
        self.next = next


def has_cycle(head: ListNode | None) -> bool:
    fast_head = head
    while head and fast_head:
        head = head.next
        fast_head = fast_head.next
        if fast_head:  # explicit check here
            fast_head = fast_head.next  # <- error here
        else:
            return False
        if head == fast_head:
            return True
    return False
```
Even the hover tooltip is correct.
![image](https://user-images.githubusercontent.com/12860284/162469198-037a0986-c270-4749-a7bf-af0c39e328ae.png)

It looks like a regression in v2022.4.0 release.

**Comment by erictraut** [comment]
Thanks for the bug report. This was a regression I introduced when adding an optimization recently. I've fixed the bug and added a new unit test based on your code sample to prevent similar future regressions.

**Comment by heejaechang** [comment]
This issue has been fixed in version 2022.4.1, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202241-13-April-2022)
